crates/ruff/src/diagnostics.rs

```diff
@@ -260,8 +260,8 @@ pub(crate) fn lint_path(
     // Lint the file.
     let (
         LinterResult {
-            data: messages,
-            error: parse_error,
+            messages,
+            has_error,
```

Comment:
Nit: maybe change to has_syntax_error ir are there other errors that set the flag to true?